src/lib.rs

```diff
@@ -76,6 +78,57 @@ struct AddressLineInfo {
     pub column: usize,
 }
 
+/// Types of symbols..
+#[derive(Clone, Copy)]
+pub enum SymbolType {
+    Unknown,
+    Function,
+    Variable,
+}
+
+/// The context of an address finding request.
+///
+/// This type passes additionl parameters to resolvers.
+#[doc(hidden)]
+pub struct FindAddrOpts {
+    /// Return the offset of the symbol from the first byte of the
+    /// object file if it is true. (False by default)
+    offset_in_file: bool,
+    /// Return the name of the object file if it is true. (False by default)
+    obj_file_name: bool,
+    /// Return the symbol(s) matching a given type. Unknown, by default, meean all types.
+    sym_type: SymbolType,
+}
```

Comment:
shouldn't it be an option whether to consult DWARF or not? It's a big performance hit to reach for DWARF and in a lot of case it might not be necessary, so let's make it an option

Reply:
Developers can use SymbolizerFeature::DebugInfoSymbols to turn it on or off. (off by default)